packages/kotti-input-number/src/InputNumber.vue

```diff
@@ -72,35 +66,72 @@ export default {
 		increaseButtonStyle() {
 			return {
 				'kt-input-number__button': true,
-				'kt-input-number__button--disabled':
-					this.currentValue == this.max || this.disabled,
+				'kt-input-number__button--disabled': this.incrementDisabled,
 			}
 		},
 		decreaseButtonStyle() {
 			return {
 				'kt-input-number__button': true,
-				'kt-input-number__button--disabled':
-					this.currentValue == this.min || this.disabled,
+				'kt-input-number__button--disabled': this.decrementDisabled,
 			}
 		},
 	},
 	watch: {
 		value(val) {
 			this.currentValue = parseFloat(val)
 		},
+		currentValue(newVal) {
+			if (!this.formError) {
+				this.$emit('input', newVal)
+			}
+			if (typeof this.min === 'number') {
+				if (newVal - this.step < this.min) {
+					this.decrementDisabled = true
+				} else if (!this.disabled) {
+					this.decrementDisabled = false
+				}
+			}
+			if (typeof this.max === 'number') {
+				if (newVal + this.step > this.max) {
+					this.incrementDisabled = true
+				} else if (!this.disabled) {
+					this.incrementDisabled = false
+				}
+			}
+		},
+		disabled(newVal) {
+			if (newVal) {
+				this.incrementDisabled = newVal
+				this.decrementDisabled = newVal
+			}
+		},
+	},
+	created() {
+		this.currentValue = typeof this.value === 'number' ? this.value : 0
 	},
 	methods: {
+		incrementValue() {
+			if (this.incrementDisabled) return
+			this.currentValue += this.step
+		},
+		decrementValue() {
+			if (this.decrementDisabled) return
+			this.currentValue -= this.step
+		},
 		/* Do not use input.stepUp - this method does not work in internet explorer and is not transpiled by Babel  */
+
 		setCurrentValue(value) {
 			if (value === this.currentValue) return
-			this.currentValue = typeof value === Number ? value : parseFloat(value)
-			this.$emit('input', this.currentValue)
+			this.currentValue =
+				typeof value === 'number' && !Number.isNaN(value)
+					? value
+					: parseFloat(value)
 		},
 	},
 }
 </script>
 <style lang="scss">
-@import '../../kotti-style/_variables.scss';
+// @import '../../kotti-style/_variables.scss';
```

Comment:
i think ur fix from the configs PR will eliminate the need for this line

Reply:
it should - but i had to comment this out for use in yoda when i copied the component, and then forgot this :)